src/CommunityToolkit.Maui.Core/Interfaces/ISaveFileDialog.shared.cs

```diff
@@ -0,0 +1,26 @@
+namespace CommunityToolkit.Maui.Core;
+
+/// <summary>
+/// Allows selecting target folder and saving files to the file system.
+/// </summary>
+public interface ISaveFileDialog
+{
+	/// <summary>
+	/// Allows selecting target folder and saving files to the file system.
+	/// </summary>
+	/// <param name="initialPath">Initial path</param>
+	/// <param name="fileName">File name with extension</param>
+	/// <param name="stream"><see cref="Stream"/></param>
+	/// <param name="cancellationToken"><see cref="CancellationToken"/></param>
+	/// <returns>True if file is saved, otherwise False</returns>
+	Task<bool> SaveAsync(string initialPath, string fileName, Stream stream, CancellationToken cancellationToken);
+
+	/// <summary>
+	/// Allows selecting target folder and saving files to the file system.
+	/// </summary>
+	/// <param name="fileName">File name with extension</param>
+	/// <param name="stream"><see cref="Stream"/></param>
+	/// <param name="cancellationToken"><see cref="CancellationToken"/></param>
+	/// <returns>True if file is saved, otherwise False</returns>
+	Task<bool> SaveAsync(string fileName, Stream stream, CancellationToken cancellationToken);
```

Comment:
Instead of returning `Task<bool>`, let's return `Task` and throw an Exception if the file is not saved.

We can create a new Exception like `FileSaveException` or something like that.

This aligns more closely to .NET MAUI's FilePicker that does things like `throws throw new PlatformNotSupportedException`.